Build failure: julia

### Steps To Reproduce

Steps to reproduce the behavior:
1. Try to update the system with nixos-rebuild
2. Build failed with message `error: builder for '/nix/store/dq3k8nz8ass49r942swi73z09a20m53r-julia-1.10.4.drv' failed with exit code 2;
       last 10 log lines:
       > -- Installing: /build/julia-1.10.4/usr/lib/libspqr.so
       > -- Installing: /build/julia-1.10.4/usr/include/SuiteSparseQR_C.h
       > -- Installing: /build/julia-1.10.4/usr/include/SuiteSparseQR_definitions.h
       > -- Installing: /build/julia-1.10.4/usr/include/SuiteSparseQR.hpp
       > -- Installing: /build/julia-1.10.4/usr/lib/cmake/SPQR/SPQRTargets.cmake
       > -- Installing: /build/julia-1.10.4/usr/lib/cmake/SPQR/SPQRTargets-release.cmake
       > -- Installing: /build/julia-1.10.4/usr/lib/cmake/SPQR/SPQRConfig.cmake
       > -- Installing: /build/julia-1.10.4/usr/lib/cmake/SPQR/SPQRConfigVersion.cmake
       > -- Installing: /build/julia-1.10.4/usr/lib/pkgconfig/SPQR.pc
       > make: *** [Makefile:79: julia-deps] Error 2
       For full logs, run 'nix log /nix/store/dq3k8nz8ass49r942swi73z09a20m53r-julia-1.10.4.drv'.`

### Build log

https://gist.github.com/thelogi/058380cd2825a9404c217c2b5eca7669

### Additional context

### Notify maintainers

@NickCao
@joshniemela
@thomasjm

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these 5 paths will be fetched (0.68 MiB download, 9.98 MiB unpacked):
  /nix/store/mhk0cm9nrjs855aq1vcivra9z9g9hn44-file-5.45
  /nix/store/5x2mvs15awr8jc1pl03cxr58c07h52dp-gnu-config-2024-01-01
  /nix/store/m9288777y854d5k0xkxgacc8rslrhb8r-gnumake-4.4.1
  /nix/store/5r0df66ikad3xw06azlqvswcvncll8wa-stdenv-linux
  /nix/store/iks1pihvbilsh5sy8qvpd638k422w9i8-update-autotools-gnu-config-scripts-hook
copying path '/nix/store/5x2mvs15awr8jc1pl03cxr58c07h52dp-gnu-config-2024-01-01' from 'https://cache.nixos.org'...
copying path '/nix/store/m9288777y854d5k0xkxgacc8rslrhb8r-gnumake-4.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/mhk0cm9nrjs855aq1vcivra9z9g9hn44-file-5.45' from 'https://cache.nixos.org'...
copying path '/nix/store/iks1pihvbilsh5sy8qvpd638k422w9i8-update-autotools-gnu-config-scripts-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/5r0df66ikad3xw06azlqvswcvncll8wa-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.40, NixOS, 24.11 (Vicuna), 24.11.20240716.ad0b5ee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/70y8v9qfn34sdfsnwh3l9772lk4rpfiy-source`

```

--- a/pkgs/development/compilers/julia/generic.nix
+++ b/pkgs/development/compilers/julia/generic.nix
@@ -14,6 +14,7 @@
 , gnum4
 , openssl
 , libxml2
+, zlib
 }:
 
 stdenv.mkDerivation rec {
@@ -40,6 +41,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     libxml2
+    zlib
   ];
 
   dontUseCmakeConfigure = true;

